kymatio/scattering1d/frontend/base_frontend.py

```diff
@@ -368,4 +435,761 @@ def _document(cls):
             n=cls._doc_array_n)
 
 
-__all__ = ['ScatteringBase1D']
+class TimeFrequencyScatteringBase1D():
+    def __init__(self, J_fr=None, Q_fr=2, F=None, average_fr=False,
+                 oversampling_fr=0, aligned=True, resample_filters_fr=True,
+                 max_pad_factor_fr=None, out_3D=False, out_type='array'):
+        self.J_fr = J_fr
+        self.Q_fr = Q_fr
+        self.F = F
+        self.average_fr = average_fr
+        self.oversampling_fr = oversampling_fr
+        self.aligned = aligned
+        if isinstance(resample_filters_fr, tuple):
+            self.resample_psi_fr, self.resample_phi_fr = resample_filters_fr
+        else:
+            self.resample_psi_fr = self.resample_phi_fr = resample_filters_fr
+        self.max_pad_factor_fr = max_pad_factor_fr
+        self.out_3D = out_3D
+        self.out_type = out_type
+
+    def build(self):
+        """Check args and instantiate `_FrequencyScatteringBase` object
+        (which builds filters).
+
+        Certain internal data, such as the amount of padding and the wavelet
+        filters to be used in the scattering transform, need to be computed
+        from the parameters given during construction. This function is called
+        automatically during object creation and no subsequent calls are
+        therefore needed.
+        """
+        # don't allow untested (and likely unneeded) combination
+        if not self.aligned and not self.out_3D:
+            raise ValueError("`aligned=False` is only allowed with `out_3D=True`")
+
+        # if config yields no second order coeffs, we cannot do joint scattering
+        if self._no_second_order_filters:
```

Comment:
where is this set?

Also, isn't the point of jtfs to have 2D filters acting on order 1?

Reply:
#729